httpx/middleware/logger.go

```diff
@@ -13,29 +13,41 @@ import (
 	"golang.org/x/net/context"
 )
 
+type loggerGenerator func(context.Context, *http.Request) logger.Logger
+
 // StdoutLogger is a logger.Logger generator that generates a logger that writes
 // to stdout.
-var StdoutLogger = stdLogger(os.Stdout)
+func StdoutLogger() loggerGenerator {
+	return stdLogger(logger.DEBUG, os.Stdout)
+}
+
+func StdoutLoggerWithLevel(lvl string) loggerGenerator {
```

Comment:
I kinda wanted to just add an argument to the `StdoutLogger()` func with level, but that would break existing callers.
Also was unsure about taking in a string here vs requiring callers to call `ParseLevel()` themselves and pass in the typed level

Reply:
aren't existing callers going to be broken anyways? you've changed it from a variable to a function.